app/src/main/java/com/alohagoha/aaaa/ui/FragmentMoviesList.kt

```diff
@@ -66,12 +61,11 @@ class FragmentMoviesList : Fragment() {
         _moviesListBinding = null
     }
 
-    override fun onDetach() {
-        super.onDetach()
-        clickListener = null
+    fun interface OnMovieCardClickListener {
+        fun onClickMovie(movieId: Int)
     }
 
-    interface OnMovieCardClickListener {
-        fun onClickMovie()
+    companion object {
+        const val FRAGMENT_TAG = "TAG:FragmentMoviesList"
     }
 }
```

Comment:
публичный/приватные методы/переменные нужно расположить в одном определенном порядке. Обычно чем публичнее, тем выше, чтобы сразу видеть внешнюю "кухню" класса, и не видеть внутреннюю

Reply:
Так ну здравый смысл мне подсказывает что companion object и interface довольно публичные. Так лучше с них начинать? Странно а я в примерах в воркшопах точно где-то видел что companion object внизу располагали) 

А еще вот наткнулся...
https://kotlinlang.org/docs/reference/coding-conventions.html#source-code-organization
 Do not sort the method declarations alphabetically or by visibility, and do not separate regular methods from extension methods. Instead, put related stuff together, so that someone reading the class from top to bottom can follow the logic of what's happening.

Но мне не хватает опыта, потому что какое-то очень общая такая рекомендация...

Reply:
Сорян, написал слишком поверхностно, предполагая, что "все всё знают".

Я не имел в виду только по публичности все отсортировать.
По ссылке как раз пишут про рекомендации порядка в классе: конструктор, переменные, методы, компаньон.
А уже каждый из блоков сортируют по видимости обычно, т.е.
public val/var
private val/var
init
public fun
private fun
companion
class